### PR TITLE
Trivial if(1) simplification

### PR DESCRIPTION
**Pull request by chmduquesne**
##### Checklist

- [ ] documentation is added or updated
- [ ] tests are added or updated

##### Description of change

The original file has a weird "if(1)" construct, which combines with a preprocessor macro. I believe that removing it makes the code more readable (and 2 lines smaller).

I think this change is trivial enough to not require me to sign the licensing agreement (let me know otherwise, I will gladly comply).

### CODE REVIEW

**Comment by mattcaswell** [comment]
IMO this change goes beyond the "fix a spelling mistake" trivial change. Some creative thought had to go into it. I think we should have a CLA for this.

File: apps/passwd.c [hunk 1]
```diff
@@ -240,23 +240,21 @@ int passwd_main(int argc, char **argv)
...
+        passwds = passwds_static;
+        if (in == NULL)
+            if (EVP_read_pw_string
+                (passwd_malloc, passwd_malloc_size, "Password: ",
+                 !(passed_salt || in_noverify)) != 0)
+                goto end;
+        passwds[0] = passwd_malloc;
+#else
             BIO_printf(bio_err, "password required\n");
             goto end;
```
Comment (mattcaswell) [inline apps/passwd.c]: Indentation of this block is now wrong.

**Comment by chmduquesne** [comment]
My bad, it looks like this construct was intended, as pointed out in http://stackoverflow.com/questions/43064381/what-is-this-c-idiom-if-1

Closing this.

Thanks!

**Comment by mattcaswell** [comment]
Well that stackoverflow question doesn't really give a justification for this usage, and personally I don't see what it is.

**Comment by chmduquesne** [comment]
It actually does, now.

**Comment by mattcaswell** [comment]
Hmmm. Well actually we regularly test all options now (we have an overnight job to do it), so avoiding bit rot is less of an issue.  I wonder what @levitte thinks.

**Comment by chmduquesne** [comment]
Just to clarify, Etienne Millon was the one who pointed this to me yesterday and I wanted an answer about why this pattern was used, hence my PR. He independently wrote the stackoverflow question and answered it himself (the reason is that it prevents bitrot by forcing the else statement to always compile). Since I was chatting with him I got the answer before he was finished writing it down.

**Comment by richsalz** [comment]
@chmduquesne , that's a great story! :)

**Comment by levitte** [comment]
So what happens if there is an option that we've guarded like this and aren't testing regularly, and one day we start doing so, does that mean we take this coding pattern away?  In other words, shall we regard them as temporary, to be taken away when we start testing them?  What happens the day we change our test and one of those options isn't tested any more, but because we don't guard the `#else` branch this way any more, we end up having bit rot in them 5 years later?  *That has already happened and is the reason this pattern was deviced*

I'm very unsure about this change.

**Comment by levitte** [comment]
Since this question seems to come up, maybe we should document this pattern a bit better than just a mere random comment...

**Comment by richsalz** [comment]
@levitte great idea.  A non-normative appendix/reference to the coding style doc?  (sorry, IETF speak...)

**Pull request closed without merging**